### PR TITLE
Submit-a-job form

### PR DESCRIPTION
**Pull request by CAYdenberg**
Hey @MarkBennett  have a quick look and see if you can live with this ... it adds a new status to Jobs: 'submitted', so the admin can easily look and see which jobs came from the web form vs which ones they put in as drafts.

I'm checking csrf tokens but as there's no authentication at this stage it might also be necessary to use a CAPTCHA? Maybe we can try it out and see how much spam comes through, but let me know your thoughts.

### CODE REVIEW

**Comment by CAYdenberg** [comment]
Keeping this as WIP because the build failed and I don't know why (yet).

**Comment by CAYdenberg** [comment]
Passed, g2g

**Comment by MarkBennett** [comment]
Hey @CAYdenberg. Will check this out. Thanks!

File: routes/index.js [hunk 1]
```diff
@@ -68,7 +69,39 @@ module.exports = (keystone) => {
 			res.render('job', job);
 		});
 	});
-	
+
+	router.post('/submit-your-job', csrfProtection, (req, res, next) => {
+		const KEYS = ['title', 'department', 'employment_type', 'company', 'location', 'summary', 'details', 'url_to_find_out_more', 'url_to_apply', 'expires_on', 'contact_email']
+		const formData = Object.assign(
+			Object.keys(req.body).filter(key => KEYS.indexOf(key) !== -1)
```
Comment (MarkBennett) [inline routes/index.js]: Cool functional approach to filtering the body for the expected keys. ;)

File: routes/index.js [hunk 2]
```diff
@@ -68,7 +69,39 @@ module.exports = (keystone) => {
...
+					acc[key] = req.body[key]
+					return acc
+				}, {}),
+			{state: 'submitted'}
+		)
+		const Job = keystone.list('Job')
+		const newJob = new Job.model(formData)
+		newJob.save(err => {
+			if (err && err.status === 403) {
+				req.error = 'Invalid token. Leave us alone.'
```
Comment (MarkBennett) [inline routes/index.js]: Take that hackers. 😎

**Comment by MarkBennett** [comment]
Sorry for the slow review. Ship it @CAYdenberg! 🚢

**Pull request merged**